### PR TITLE
gnrc_sixlowpan_frag: Initial import of a fragment size hint feature

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This allows for sending of fragments smaller than the restrictions set by the link-layer PDU. E.g. to put some slack for IPHC into the first fragment (see https://tools.ietf.org/html/draft-ietf-6lo-fragment-recovery-02#section-4.1).

### Testing procedure
Currently there is no test for this (I will provide a test application for my [minimal fragment forwarding](https://tools.ietf.org/html/draft-ietf-6lo-minimal-fragment-00) implementation that also tests this feature), so just compiling `gnrc_networking` with the `gnrc_sixlowpan_frag_hint` pseudo-module for a 6Lo-capable board (e.g. `samr21-xpro`) should suffice (this will not be done by the CI!).

### Issues/PRs references
Depends on #11021 (mainly because there is a lot of conflict between those two, but functionally it is independent from that PR).

Route to 6Lo minimal fragment forwarding

### CODE REVIEW

**Comment by miri64** [comment]
#11068 is to experimental so it doesn't make any sense to try to get this into the release => post-poned.

**Comment by miri64** [comment]
Rebased however to resolve conflict with current master

**Comment by miri64** [comment]
Rebased to current master and dependencies

**Comment by miri64** [comment]
Rebased to current master and dependencies.

**Comment by miri64** [comment]
Rebased to current master.

File: sys/include/net/gnrc/sixlowpan/frag.h [hunk 1]
```diff
@@ -94,6 +97,13 @@ typedef struct {
     uint16_t tag;           /**< Tag used for the fragment */
     uint16_t offset;        /**< Offset of the Nth fragment from the beginning of the
                              *   payload datagram */
+#ifdef MODULE_GNRC_SIXLOWPAN_FRAG_HINT
+    /**
+     * @brief   Hint for the size (smaller than link-layer PDU) of the
+     *          *first fragment*
```
Comment (cgundogan) [inline sys/include/net/gnrc/sixlowpan/frag.h]: Why **first** fragment? The code seems to indicate something else (for all fragments?). Also, other parts of the documentation also refer to more fragments than just the first.

File: sys/include/net/gnrc/sixlowpan/frag/hint.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_sixlowpan_frag_hint    Fragment size hint
+ * @ingroup     net_gnrc_sixlowpan_frag
+ * @brief       Provides a hint for smaller fragment sizes than the link-layer
+ *              PDU for the *first fragment*.
```
Comment (cgundogan) [inline sys/include/net/gnrc/sixlowpan/frag/hint.h]: As above, why **first**?

File: sys/include/net/gnrc/sixlowpan/frag.h (hunk 1 above)
Comment (miri64) [reply sys/include/net/gnrc/sixlowpan/frag.h]: I think I meant the first fragment that is handled by this... But I just remove it, as it is indeed confusing.

**Comment by miri64** [comment]
Reworded the unclear lines.

**Comment by miri64** [comment]
Sorry, accidentally pushed an old version -.-

**Comment by miri64** [comment]
(it is fixed now)

File: sys/include/net/gnrc/sixlowpan/frag.h [hunk 3]
```diff
@@ -94,6 +97,13 @@ typedef struct {
     uint16_t tag;           /**< Tag used for the fragment */
     uint16_t offset;        /**< Offset of the Nth fragment from the beginning of the
                              *   payload datagram */
+#ifdef MODULE_GNRC_SIXLOWPAN_FRAG_HINT
+    /**
+     * @brief   Hint for the size (smaller than link-layer PDU) of for the next
```
Comment (cgundogan) [inline sys/include/net/gnrc/sixlowpan/frag.h]: of for? (:

**Pull request merged**